_api_app/app/Sites/Sections/SiteSectionsDataService.php

```diff
@@ -555,6 +557,108 @@ public function galleryOrder($name, $new_files)
         return ['error_message' => 'Section "' . $name . '" not found!'];
     }
 
+    public function galleryUpload($data)
+    {
+        $path = $data['path'];
+        $file = $data['value'];
+        $section_idx = explode('/', $path)[2];
+
+        if (!$file->isValid()) {
+            return [
+                'status' => 0,
+                'error' => 'Upload failed.'
+            ];
+        }
+
+        $validator = Validator::make($data, [
+            'value' => 'max:' .  config('app.image_max_file_size') . '|mimetypes:' . implode(',', config('app.image_mimetypes'))
+        ]);
+
+        if ($validator->fails()) {
+            return [
+                'status' => 0,
+                'error' => implode(' ', $validator->messages()->all())
+            ];
+        }
+
+        if (ImageHelpers::isCorruptedImage($file)) {
```

Comment:
Can this be implemented as part of the `Validator` (https://laravel.com/docs/5.2/validation#custom-validation-rules)